Based on https://lists.xenproject.org/archives/html/xen-devel/2020-09/msg00478.html

[This patch should be replaced with the committed version, when it's
available from the Xen git repository.]

--- a/xen/arch/arm/Rules.mk	2020-05-14 14:19:32.000000000 +0200
+++ b/xen/arch/arm/Rules.mk	2020-09-10 11:17:53.640315194 +0200
@@ -17,6 +17,7 @@
 
 CFLAGS-$(CONFIG_ARM_64) += -mcpu=generic
 CFLAGS-$(CONFIG_ARM_64) += -mgeneral-regs-only # No fp registers etc
+$(call cc-option-add,CFLAGS-$(CONFIG_ARM_64),CC,-mno-outline-atomics)
 
 ifneq ($(call cc-option,$(CC),-fvisibility=hidden,n),n)
 CFLAGS += -DGCC_HAS_VISIBILITY_ATTRIBUTE
